From 48eb807d543955526ec2ae0546a0904176315e55 Mon Sep 17 00:00:00 2001
From: Marian Buschsieweke <marian.buschsieweke@posteo.net>
Date: Sun, 11 Jan 2026 09:49:02 +0100
Subject: [PATCH] cmake: generate pkg-config file nng.pc using CMake

The pkg-config file allows consumers such as KiCad to discover nng.
---
 CMakeLists.txt  |  2 ++
 cmake/nng.pc.in | 12 ++++++++++++
 2 files changed, 14 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1f5c49f4..fedd37af 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -324,6 +324,8 @@ set(CPACK_SOURCE_PACKAGE_FILE_NAME
 set(CPACK_RESOURCE_FILE_LICENSE ${CMAKE_CURRENT_SOURCE_DIR}/LICENSE.txt)
 set(CPACK_PACKAGE_INSTALL_DIRECTORY "nng")
 set(CPACK_PACKAGE_FILE_NAME "${PROJECT_NAME}-v${NNG_PACKAGE_VERSION}")
+configure_file(cmake/nng.pc.in nng.pc @ONLY)
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/nng.pc DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig/)
 
 add_custom_target(dist COMMAND ${CMAKE_MAKE_PROGRAM} package_source)
 include(CPack)
diff --git a/cmake/nng.pc.in b/cmake/nng.pc.in
new file mode 100644
index 00000000..5996fd29
--- /dev/null
+++ b/cmake/nng.pc.in
@@ -0,0 +1,12 @@
+# This template is used to generate nng.pc using CMake's
+# configure_file(... @ONLY) using CMake variables.
+prefix="@CMAKE_INSTALL_PREFIX@"
+libdir=@CMAKE_INSTALL_FULL_LIBDIR@
+includedir=@CMAKE_INSTALL_FULL_INCLUDEDIR@
+
+Name: @CPACK_PACKAGE_NAME@
+Version: @CPACK_PACKAGE_VERSION@
+Description: @CPACK_PACKAGE_DESCRIPTION_SUMMARY@
+URL: https://@CPACK_PACKAGE_VENDOR@
+Cflags: -I"${includedir}"
+Libs: -L"${libdir}" -lnng
